File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int ret = 0;
        for (int i = 0; i < k; ++i) {
            if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u') {
                ret++;
            }
        }
        int cur = ret;
        for (int i = k; i < s.length(); ++i) {
            if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u') {
                cur++;
            }
            if (s[i - k] == 'a' || s[i - k] == 'e' || s[i - k] == 'i' || s[i - k] == 'o' || s[i - k] == 'u') {
                cur--;
            }
            ret = max(cur, ret);
        }
        return ret;
    }
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def isVowel(self, char):
        return char == 'a' or char == 'e' or char == 'i' or char == 'o' or char == 'u'
    
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        numVowels, curVowels = 0, 0
        for i in range(len(s)):
            if i < k and self.isVowel(s[i]):
                numVowels += 1
                curVowels += 1
            elif i >= k:
                if self.isVowel(s[i]):
                    curVowels += 1
                if self.isVowel(s[i - k]):
                    curVowels -= 1
                numVowels = max(numVowels, curVowels)
        return numVowels
                
        
        